File: Data-Mining/assignment3/knn/show_image.m
function show_image(X)
%SHOW_IMAGE Show a CAPTCHA image
%   Inputs:
%       X: 144x4 matrix, 4 digits in an image, each digit is a 144x1 vector.

num = size(X, 2);
X = reshape(X, 12, 12 * num);
imagesc(X); colormap(gray); axis image;
end

File: Data-Mining/assignment3/knn/show_image.py
import matplotlib.pyplot as plt


def show_image(x):
    """
    Inputs:
        x: (N, 144) matrix, N digits in an image, each digit is a (144, ) column vector.
    """
    num = x.shape[0]
    x = x.reshape(num, 12, 12).transpose(1, 0, 2).reshape(12, num * 12)
    plt.imshow(x, cmap='gray')
